services/getstream/unfollowFeed.js

```diff
@@ -0,0 +1,20 @@
+const {unFollowFeedProcessQueue} = require('../score/queueSenderForRedis');
+
+module.exports = async (feedName, userId, targetFeed, unfollowUserId) => {
+  try {
+    if (!feedName || !userId || !targetFeed || !unfollowUserId) {
+      throw new Error('Invalid parameters');
+    }
+    const result = await unFollowFeedProcessQueue.add({
+      feedName,
+      userId,
+      targetFeed,
+      unfollowUserId
+    });
+    return result;
+  } catch (error) {
+    return {
+      error: error.message
+    };
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new function `unfollowFeed` seems to be well implemented. It validates the parameters before adding a task to the queue, and it handles errors by returning them in an object. However, it would be better if the error handling was more robust. Instead of just returning the error message, you could also log the error for debugging purposes. Here's how you can do it:

```diff
  catch (error) {
+   console.error(error);
    return {
      error: error.message
    };
  }
```
This way, if an error occurs, you will have more information about it in your logs, which can help with troubleshooting.

Also, consider using a more specific error message in line 6. "Invalid parameters" is quite generic and might not provide enough information for debugging if an error occurs. You could check each parameter individually and throw a specific error message for each one. This would make it easier to identify which parameter is causing the issue.

```diff
- if (!feedName || !userId || !targetFeed || !unfollowUserId) {
-   throw new Error('Invalid parameters');
- }
+ if (!feedName) {
+   throw new Error('Invalid feedName');
+ }
+ if (!userId) {
+   throw new Error('Invalid userId');
+ }
+ if (!targetFeed) {
+   throw new Error('Invalid targetFeed');
+ }
+ if (!unfollowUserId) {
+   throw new Error('Invalid unfollowUserId');
+ }
```

These changes should improve the maintainability and debuggability of your code.


<!-- This is an auto-generated comment by OSS CodeRabbit -->